File: frontend/src/services/books.ts
import api from "./api";
import type { Book } from "../types/book";

export async function getBooks(): Promise<Book[]> {
  const response = await api.get("/books");
  return response.data;
}

export async function getBook(id: number): Promise<Book> {
  const response = await api.get(`/books/${id}`);
  return response.data;
}

File: backend/app/routes/books.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.db.database import SessionLocal
from app.models.book import Book

router = APIRouter()


def get_db():
    db = SessionLocal()
    try:
        yield db
    finally:
        db.close()


@router.get("/books")
def get_books(db: Session = Depends(get_db)):
    books = db.query(Book).all()
    return books

@router.get("/books/{book_id}")
def get_book(book_id: int, db: Session = Depends(get_db)):
    book = db.query(Book).filter(Book.id == book_id).first()

    if not book:
        return {"error": "Book not found"}

    return book